src/services/download/mod.rs

```diff
@@ -206,7 +202,7 @@ fn download_future_stream(
 ///
 /// This is common functionality used by all many downloaders.
 async fn download_stream(
-    source: SourceFileId,
+    source: &str,
     stream: impl Stream<Item = Result<Bytes, DownloadError>>,
```

Comment:
_nit_: Just noticing this. Rather than making this `impl Stream`, can we make this a type parameter on the function? That should allow for proper propagation of `Send` and the likes in all cases.